app/src/androidTest/java/com/urbandroid/sleep/addon/stats/model/socialjetlag/SocialJetlagStatsTest.java

```diff
@@ -71,4 +76,169 @@ public void testGetRecordIrregularity() {
 
         assertThat(stats.getRecordIrregularity(record)).isEqualTo( ((6.625f - 3.5f) + (8.5f - 6.5f)) / 2f );
     }
+
+    @Test
+    public void calculateSRIEdgeCases() {
+        SocialJetlagStats  stats = new SocialJetlagStats(cr, false);
+
+        // one bitset empty
+        BitSet day1Interval = new BitSet(1440);
+        BitSet day2Interval = new BitSet();
+        day1Interval.set(0,720);
+        Assert.assertEquals(1.0, stats.calculateSRI(day1Interval, day2Interval), .0001);
+
+        // awake 24 hours
+        BitSet interval3 = new BitSet(1440);
+        BitSet interval4 = new BitSet(1440);
+        interval3.set(0,1440);
+        Assert.assertEquals(0, stats.calculateSRI(interval3, interval4), .0001);
+
+        // awake 0 hours
+        BitSet interval5 = new BitSet(1440);
+        BitSet interval6 = new BitSet(1440);
+        Assert.assertEquals(1.0, stats.calculateSRI(interval5, interval6), .0001);
+    }
+
+    @Test
+    public void testCalculateSRI() {
+        BitSet day1Interval = new BitSet(1440);
+        BitSet day2Interval = new BitSet(1440);
+        day1Interval.set(0,720);
+        day2Interval.set(0,720);
+        // testing for when the intevals are the same
+        SocialJetlagStats  stats = new SocialJetlagStats(cr, false);
+        float sameInterval = stats.calculateSRI(day1Interval, day2Interval);
+        Assert.assertEquals(1.0, sameInterval, .0001);
+
+        // general test, one interval enclosed inside the other
+        BitSet day3Interval = new BitSet(1440);
+        day3Interval.set(0,1440);
+        Assert.assertEquals(0.5, stats.calculateSRI(day1Interval, day3Interval), .0001);
+        Assert.assertEquals(0.5, stats.calculateSRI(day2Interval, day3Interval), .0001);
+
+        // testing when intervals are completely different
+        BitSet day4Interval = new BitSet(1440);
+        day4Interval.set(720, 1440);
+        Assert.assertEquals(0, stats.calculateSRI(day1Interval, day4Interval), .0001);
+
+        // testing when there is some overlap
+        BitSet interval5 = new BitSet(1440);
+        BitSet interval6 = new BitSet(1440);
+        interval5.set(0, 720);
+        interval6.set(360, 1440);
+        Assert.assertEquals(0.25, stats.calculateSRI(interval5, interval6), .0001);
+
+        // single overlap
+        BitSet interval7 = new BitSet(1440);
+        BitSet interval8 = new BitSet(1440);
+        interval7.set(0, 2);
+        interval8.set(1, 2);
+        Assert.assertEquals(1 - 1.0/1440.0, stats.calculateSRI(interval7, interval8), .000001);
+
+        // pretty regular
+        BitSet interval9 = new BitSet(1440);
+        BitSet interval10 = new BitSet(1440);
```

Comment:
i think what we should do is instead of doing -200 store it in a variable 
we should be able to deduce how the value 200 came from and from what variables